pyfr/plugins/tavg.py

```diff
@@ -338,4 +346,147 @@ def __call__(self, intg):
 
                     self.tstart_acc = intg.tcurr
 
-                self.tout_last = intg.tcurr
\ No newline at end of file
+                self.tout_last = intg.tcurr
+
+
+class TavgCLIPlugin(CLIBasePlugin):
+    name = 'tavg'
+    systems = ['*']
+    formulations = ['dual', 'std']
+    cli = [
+        ('merge', 'merge_cli', ('solns', {'nargs': '*', 'help': 'Averages to merge'}), 
+                               ('output', {'help': 'Output file name'})),
+    ]
+
+    def merge_cli(self, args):
+        self.args = args
+        start_time = np.empty((0, 1))
+        end_time = np.empty((0, 1))
+
+        active = self._merge_active_files(args.solns)
+        merged_times = [(r.start, r.end) for r, _, _ in active]
+        std_all = all([s == 'all' for _, _, s in active])
```

Comment:
For this kind of thing I'd be inclined to require that every section in the config file other than nsteps, averaging mode, and dt out need to be the same.  (So for each file grab the cfg items for our plugins section, pop the aforementioned keys, and then require them all to be the same.)